File: add_credits.js
const { DynamoDBClient, ScanCommand, UpdateItemCommand } = require('/var/www/nexus-ai-pro/node_modules/@aws-sdk/client-dynamodb');
const client = new DynamoDBClient({ region: 'ap-south-1' });

async function run() {
  let users = [];
  let lastKey = undefined;
  do {
    const res = await client.send(new ScanCommand({
      TableName: 'nexus-ai-pro-users',
      ProjectionExpression: 'userId',
      ExclusiveStartKey: lastKey
    }));
    users.push(...res.Items);
    lastKey = res.LastEvaluatedKey;
  } while (lastKey);
  
  console.log('Found ' + users.length + ' users');
  let success = 0, errors = 0;
  
  for (let i = 0; i < users.length; i++) {
    try {
      await client.send(new UpdateItemCommand({
        TableName: 'nexus-ai-pro-users',
        Key: { userId: users[i].userId },
        UpdateExpression: 'ADD credits :c',
        ExpressionAttributeValues: { ':c': { N: '45' } }
      }));
      success++;
    } catch(e) {
      console.log('Error:', users[i].userId.S, e.message);
      errors++;
    }
    if ((i+1) % 100 === 0) console.log('Progress: ' + (i+1) + '/' + users.length);
  }
  console.log('Done! Success: ' + success + ', Errors: ' + errors);
}
run();

File: add_credits.py
import boto3

client = boto3.client('dynamodb', region_name='ap-south-1')
paginator = client.get_paginator('scan')

ids = []
for page in paginator.paginate(TableName='nexus-ai-pro-users', ProjectionExpression='userId'):
    ids.extend(item['userId']['S'] for item in page['Items'])

print(f'Found {len(ids)} users')

success = 0
failed = 0

for i, uid in enumerate(ids):
    try:
        client.update_item(
            TableName='nexus-ai-pro-users',
            Key={'userId': {'S': uid}},
            UpdateExpression='ADD credits :c',
            ExpressionAttributeValues={':c': {'N': '45'}}
        )
        success += 1
    except Exception as e:
        failed += 1
        print(f'FAILED: {uid} - {e}')
    if (i+1) % 100 == 0:
        print(f'Progress: {i+1}/{len(ids)}')

print(f'Done! Success: {success}, Failed: {failed}')
